### PR TITLE
[CELEBORN-823][DOC] Add Celeborn architecture document

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
As title.

### Why are the changes needed?
As title.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
No.

### CODE REVIEW

**Comment by waitinfuture** [comment]
cc @pan3793

File: docs/developers/overview.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+Currently, Celeborn only supports `Local Disks` and `HDFS`. Supporting for other storage systems are under working.
+
+## Compute Engine Integration
+Celeborn's primary components(i.e. Master, Worker, Client) are engine irrelevant. The Client APIs are extensible
+and easy to implement plugins for various engines.
+
+Currently, Celeborn officially supports Spark(both Spark 2.x and Spark 3.x), Flink(1.14/1.15/1.17), and
+[Gluten](https://github.com/oap-project/gluten). Also developers are integrating Celeborn with other engines,
+for example [MR3](https://github.com/mr3project/hive-mr3).
```
Comment (pan3793) [inline docs/developers/overview.md:80]: Maybe this link is more accurate

```suggestion
for example [MR3](https://mr3docs.datamonad.com/docs/mr3/).
```

File: docs/developers/overview.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Celeborn community is also working on integrating Celeborn with other engines.
+
+## Graceful Upgrade
```
Comment (pan3793) [inline docs/developers/overview.md:84]: Maybe "rolling upgrade" or "graceful shutdown"?

File: docs/developers/overview.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+# Celeborn Architecture
+
+This articles introduces high level Celeborn Architecture. For more detailed description of each module/process,
```
Comment (pan3793) [inline docs/developers/overview.md:19]:
```suggestion
This article introduces high level Apache Celeborn(Incubating) Architecture. For more detailed description of each module/process,
```

File: docs/developers/overview.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+Celeborn stores shuffle data in configurable multiple layers, i.e. `Memroy`, `Local Disks`, `Distributed File System`,
+and `Object Store`. Users can specify any combination of the layers on each Worker.
+
+Currently, Celeborn only supports `Local Disks` and `HDFS`. Supporting for other storage systems are under working.
+
+## Compute Engine Integration
+Celeborn's primary components(i.e. Master, Worker, Client) are engine irrelevant. The Client APIs are extensible
+and easy to implement plugins for various engines.
+
+Currently, Celeborn officially supports Spark(both Spark 2.x and Spark 3.x), Flink(1.14/1.15/1.17), and
```
Comment (AngersZhuuuu) [inline docs/developers/overview.md:89]: Also add a link to spark and flink? such as [Spark](https://spark.org.apache)

File: docs/developers/overview.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+[Gluten](https://github.com/oap-project/gluten). Also developers are integrating Celeborn with other engines,
+for example [MR3](https://github.com/mr3project/hive-mr3).
+
+Celeborn community is also working on integrating Celeborn with other engines.
+
+## Graceful Upgrade
+In order not to impact running applications when upgrading Celeborn Cluster, Celeborn implements Graceful Upgrade.
```
Comment (AngersZhuuuu) [inline docs/developers/overview.md:96]: Add a link to  `Graceful Upgrade`.?

**Comment by pan3793** [comment]
is svg editable? usually we should provide the editable source file of image too

**Comment by waitinfuture** [comment]
ping @pan3793 @AngersZhuuuu comments addressed, please take a look again

**Comment by waitinfuture** [comment]
> 



> is svg editable? usually we should provide the editable source file of image too

I'm not sure about this, I just followed the `docs/assets/logo/celeborn-*.svg` files. They are relative small.

File: docs/developers/overview.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+![ESS](/assets/docs/ess.svg)
+
+Besides inefficiency, traditional shuffle framework requires large local storage in compute node to store shuffle
+data, thus blocks the adoption of disaggregated architecture.
+
+Apache Celeborn(Incubating) solves the problems by reorganizing shuffle data in a more efficient way, and storing the data in
+a separate service. The high level architecture of Celeborn is as follows:
+
+![RSS](/assets/docs/rss.svg)
```
Comment (cfmcgrady) [inline docs/developers/overview.md:34]: nit: rss.svg -> celeborn.svg
```suggestion
![Celeborn](/assets/docs/celeborn.svg)
```

**Review by cfmcgrady — APPROVED** [review]

**Comment by pan3793** [comment]
> ... They are relative small.

@waitinfuture I'm not concern the size, I mean we should provide the "editable source file" of image in case we want to modify it in the future. how do you generate the svg?

**Review by pan3793 — APPROVED** [review]
LGTM. except the source file of svg

File: docs/assets/docs/Untitled Diagram.drawio
Comment (pan3793) [inline docs/assets/docs/Untitled Diagram.drawio:1]: Please rename and move to `docs/assets/img`
```
docs/assets/img/celeborn.drawio
docs/assets/img/celeborn.svg
```

Comment (waitinfuture) [reply docs/assets/docs/Untitled Diagram.drawio:1]: done

**Comment by waitinfuture** [comment]
> > ... They are relative small.
> 
> @waitinfuture I'm not concern the size, I mean we should provide the "editable source file" of image in case we want to modify it in the future. how do you generate the svg?

I replaces the files with editable svg format, drawn in https://app.diagrams.net/

**Comment by waitinfuture** [comment]
Thanks, merging to main.

**Pull request closed without merging**